lib/frrscript.c

```diff
@@ -32,6 +32,84 @@
 
 DEFINE_MTYPE_STATIC(LIB, SCRIPT, "Scripting");
 
+/*
+ * Script name hash utilities
+ */
+
+struct frrscript_names_head frrscript_names_hash;
+
+/*
+ * Wrapper for frrscript_names_add
+ * Use this to register hook calls when a daemon starts up
+ */
+int frrscript_names_add_function_name(const char *function_name)
+{
+	struct frrscript_names_entry *insert =
+		XCALLOC(MTYPE_TMP, sizeof(*insert));
```

Comment:
This should be a new `MTYPE`, not `MTYPE_TMP`

Reply:
updated to `MTYPE_SCRIPT` which is already an MTYPE for frrscript - let me know if I should make a whole new one instead